### PR TITLE
Reorganised folders, Started cpu experiment metrics

### LINKED ISSUE #19: Add analysers to calculate Integral of resource usage
**Some details about the data we are going to integrate:**

The data are sampled every second.

**Some references:**
- Different type of integrating sampled data: http://dmpeli.math.mcmaster.ca/Matlab/Math4Q3/NumMethods/Lecture3-4.html. The Trapezoidal rule seems suited for our case. 
- A python library enabling the computation of integrals: http://docs.scipy.org/doc/scipy-0.14.0/reference/integrate.html

### LINKED ISSUE #21: The Mode can be represented by different values
Reference: http://mathworld.wolfram.com/Mode.html

We should update the scripts and the [database schema](https://github.com/benchflow/analysers/issues/6).

### LINKED ISSUE #25: The mode has a wrong value
The `mode` has a wrong value. You wrongly assign the `max` value as you can see:
- https://github.com/benchflow/analysers/blob/dev/analysers/cpu.py#L44
- https://github.com/benchflow/analysers/blob/dev/analysers/ram.py#L44
- https://github.com/benchflow/analysers/blob/dev/analysers/processDuration.py#L44

### PR DESCRIPTION
**Pull request by Cerfoglg**
Organised folders for trials and experiments scripts. Added basic experiment metrics for cpu.

Works on #28, #22 
Fixes #25 
Fixes #21 
Fixes #19

### CODE REVIEW

File: analysers/experiment_analysers/cpu.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+modeMax = CassandraRDD.select("cpu_mode") \
+    .where("experiment_id=?", experimentID) \
+    .map(lambda x: (max(x["cpu_mode"]), 0)) \
+    .sortByKey(0, 1) \
+    .map(lambda x: x[0]) \
+    .first()
+
+
+# TODO: Fix this
+query = [{"experiment_id":experimentID, "cpu_mode":[modeMin, modeMax], "cpu_median":[medianMin, medianMax], \
```
Comment (VincenzoFerme) [inline analysers/experiment_analysers/cpu.py]: I don't see an update to the Cassandra schema to store these data: https://github.com/benchflow/benchflow/pull/7/files

**Comment by VincenzoFerme** [comment]
@Cerfoglg please map in the description, the issues you worked on and the ones that are fixed by this pull request.

File: analysers/experiments/cpu.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+from pyspark_cassandra import CassandraSparkContext
+from pyspark_cassandra import RowFormat
+from pyspark import SparkConf
+
+# Takes arguments: Spark master, Cassandra host, Minio host, path of the files
+sparkMaster = sys.argv[1]
+cassandraHost = sys.argv[2]
+trialID = sys.argv[3]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
```
Comment (VincenzoFerme) [inline analysers/experiments/cpu.py:17]: `cassandraKeyspace`, `srcTable`, `destTable` should be defined as part of the configuration file and passed as parameters. The same applies also to other scripts.

File: analysers/experiments/cpu.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+def sortAndGet(field, asc):
+    v = CassandraRDD.select(field) \
+        .where("experiment_id=?", experimentID) \
+        .map(lambda x: (x[field], 0)) \
+        .sortByKey(asc, 1) \
+        .map(lambda x: x[0]) \
+        .first()
+    return v
+
+dataMin = sortAndGet("cpu_min", 1)
```
Comment (VincenzoFerme) [inline analysers/experiments/cpu.py:42]: Why can't you just sort once and get all the data you need out of the sorted RDD?

File: analysers/trials/cpu.py [hunk 4]
```diff
@@ -29,9 +32,26 @@
```
Comment (VincenzoFerme) [inline analysers/trials/cpu.py:33]: You should apply caching and configuration passed from the spark-tasks-sender also to the analysers script working on trials, and in general to all the scripts.

File: analysers/trials/cpu.py [hunk 5]
```diff
@@ -29,9 +32,26 @@
...
+mode = list()
+highestCount = data[0][0]        
+for d in data:
+    if d[0] == highestCount:
+        mode.append(d[1])
     else:
-        return (r['cpu_percent_usage'], 0)
+        break
 
 data = sc.cassandraTable(cassandraKeyspace, srcTable) \
```
Comment (VincenzoFerme) [inline analysers/trials/cpu.py]: Why do you need to perform another query to Cassandra?

File: analysers/trials/cpu.py [hunk 6]
```diff
@@ -77,12 +97,15 @@ def computeMedian(d):
 marginError = stdE * 2
 CILow = mean - marginError
 CIHigh = mean + marginError
+CI = [CILow, CIHigh]
```
Comment (VincenzoFerme) [inline analysers/trials/cpu.py]: Lets store them as two separated values

File: analysers/trials/dumpSize.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+from pyspark import SparkConf
+
+# Takes arguments: Spark master, Cassandra host, Minio host, path of the files
+sparkMaster = sys.argv[1]
+cassandraHost = sys.argv[2]
+trialID = sys.argv[3]
+minioHost = sys.argv[4]
+filePath = sys.argv[5]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
```
Comment (VincenzoFerme) [inline analysers/trials/dumpSize.py]: `cassandraKeyspace`, `table`, `minioPort` should be defined as part of the configuration file and passed as parameters. The same applies also to other scripts.

File: analysers/trials/processDuration.py [hunk 8]
```diff
@@ -5,6 +5,9 @@
 import uuid
 import math
```
Comment (VincenzoFerme) [inline analysers/trials/processDuration.py:7]: All the comments made for `analysers/cpu.py` apply

Comment (VincenzoFerme) [inline analysers/trials/processDuration.py]: All the comments made for `analysers/cpu.py` apply

File: analysers/trials/processDuration.py [hunk 9]
```diff
@@ -77,12 +97,15 @@ def computeMedian(d):
 marginError = stdE * 2
 CILow = mean - marginError
 CIHigh = mean + marginError
+CI = [CILow, CIHigh]
+
+dataIntegral = sum(integrate.cumtrapz(data))[0]
```
Comment (VincenzoFerme) [inline analysers/trials/processDuration.py]: We don't need it

File: analysers/trials/ram.py [hunk 10]
```diff
@@ -29,9 +29,26 @@
```
Comment (VincenzoFerme) [inline analysers/trials/ram.py:33]: All the comments made for `analysers/cpu.py` apply

File: analysers/experiments/cpu.py (hunk 2 above)
Comment (Cerfoglg) [inline analysers/experiments/cpu.py:17]: @VincenzoFerme This ties into the spark tasks sender issue of reorganising the way arguments are passed to the scripts

Comment (VincenzoFerme) [inline analysers/experiments/cpu.py:17]: @Cerfoglg update that issue to point out you also need to update these scripts.

**Pull request merged**